lib/blockchain/chain.js

```diff
@@ -2006,6 +2016,36 @@ class Chain extends AsyncEmitter {
       }
     }
 
+    // Transactions are not allowed in any block before
+    // a certain amount of chainwork has been accumulated.
+    // This a non-malleated, permanently invalid block
+    // and whoever sent it should be banned.
+    if (prev.height + 1 < this.network.txStart) {
+      let invalid = false;
+
+      // No transactions allowed besides coinbase
+      if (block.txs.length > 1)
+        invalid = true;
+
+      // No claims or airdrops allowed in coinbase yet.
+      const cb = block.txs[0];
+      if (cb.outputs.length > 1)
+        invalid = true;
+
+      // Sanity check
+      if (!cb.outputs[0].covenant.isNone())
+        invalid = true;
+
+      if (invalid) {
+        this.setInvalid(block.hash());
```

Comment:
Good call on setting the block as invalid here, because it is normally called further down the call stack in a try/catch around `chain.setInvalid` :+1: